common/usermode.c

```diff
@@ -135,6 +135,54 @@ static inline long __user_text syscall(long syscall_nr, long arg1, long arg2, lo
     return return_code;
 }
 
+static inline long __user_text _sysenter(long syscall_nr, long arg1, long arg2, long arg3,
+                                         long arg4, long arg5) {
+    register long return_code asm(STR(_ASM_AX));
+    register long _arg4 asm("r8") = arg4;
+    register long _arg5 asm("r9") = arg5;
+    register long _arg3 asm("r10") = arg3;
+
+    /* clang-format off */
+    asm volatile (
+        SAVE_CALLEE_SAVED_REGS()
+        "mov %%" STR(_ASM_SP) ", %%" STR(_ASM_CX) "\n"
+        "lea 1f(%%"STR(_ASM_IP)"), %%" STR(_ASM_DX) "\n"
+        "sysenter\n"
+        "1: " RESTORE_CALLEE_SAVED_REGS()
+        : "=a"(return_code)
+        : "0"(syscall_nr), "S"(arg1), "D" (arg2), "r" (_arg3), "r"(_arg4), "r"(_arg5)
+        : STR(_ASM_CX), STR(_ASM_DX)
+    );
+    /* clang-format on */
+
+    return return_code;
+}
+
+static __user_bss enum syscall_mode sc_mode = SYSCALL_MODE_SYSCALL;
+
+bool __user_text syscall_mode(enum syscall_mode mode) {
+    if (mode < SYSCALL_MODE_SYSCALL || mode > SYSCALL_MODE_INT) {
```

Comment:
enum is `unsigned int`, so for this check to work the SYSCALL_MODE_SYSCALL should start at 1

Reply:
We shouldn't change SYSCALL_MODE_SYSCALL but leave the code as-is -- kind of defensive programming.

Reply:
only checking upper bound instead. good so?